### ISSUE TITLE
Firewall setup (allow SSH) for windscribe-CLI

### ISSUE THREAD

**Issue by ekasprzak** [Windscribe/Desktop-App#200]
**Is your feature request related to a problem? Please describe.**
I can't find any documentation on how to setup windscribe firewall for CLI version (how to allow SSH connection).

**Describe the solution you'd like**
Configure windscribe firewall to allow SSH connection

**Describe alternatives you've considered**
Using old, unmaintained, unsecure CLI version.

**Additional context**
I'm using vagrant to manage my virtual machines, on one of those I use windscribe (cli).
Still using the old, long time unmaintained version, because there windscribe connect (which turns on firewall) doesn't block me from accessing it via vagrant (which is SSH).
Using new CLI version (just tried 2.12.7) when I connect to windscribe, my SSH connection is immediately interrupted.
I guess this is because firewall default conf which works differently than the one in legacy CLI.

**Comment by blknight88** [comment]
I have the same issue, I assume this is cause the "lanbypass" option is gone, this exists in the old CLI client and allows LAN connections to go through when the Firewall is On, however that seems to be missing in the new client. 

Is that going to be added?

**Comment by jaxu** [comment]
There's a configuration file at `~/.config/Windscribe/windscribe_cli.conf` that contains almost everything that the GUI supports.  You can toggle `AllowLANTraffic` in there to `true`, and then reload the config with `windscribe-cli preferences reload`

**Comment by blknight88** [comment]
@jaxu thanks for the suggestion, I was able to get it to work by editing that file with that change after my first login. 

However now I have a new issue where it won't auto connect on reboot. I tried setting "Autoconnect=true" but every time after reboot it shows logged in but connection state as disconnected. Am I missing something?

I tried looking for documentation on this file in case there is something else in here I need to change but couldn't find any so far.

**Comment by ekasprzak** [comment]
@jaxu great, thank you!

@blknight88 can you point me to the documentation for this file, please?

BTW, I think the no autoconnect after reboot problem (which I also experience now after setting AllowLANTraffic) deserves another issue to be opened.

**Comment by blknight88** [comment]
@ekasprzak I couldn't find any documentation on this file unfortunately 

@jaxu I did miss that the first time and it is pretty obvious as if I go back to the file my changes won't be there, however on subsequent attempts I have confirmed it did save my settings however it still does not auto connect on reboot. 

My process being the following: 

`nano ~/.config/Windscribe/windscribe_cli.conf

  #Modify these values 
  AllowLANTraffic=true
  Autoconnect=true

windscribe-cli preferences reload`

In the old version, the 1.4 CLI I only got this to work by making a systemd windscribe CLI service, not sure if that is needed again here? If so not sure how it would need to be configured to make it work.

**Comment by jaxu** [comment]
No, it should just work with that.  What is the output of `windscribe-cli status` after reboot? I don't have enough information to debug this with you.  Please send in a debug log and DM me your username.

**Comment by blknight88** [comment]
@jaxu Here is the output of `windscribe-cli status`

```
Internet connectivity: available
Login state: Logged in
Firewall state: Off
Connect state: Disconnected
Data usage: 75.63 GB / Unlimited
```

Debug logs sent, however I am not sure how I can DM you in Github?

**Comment by jaxu** [comment]
Hah, sorry, wrong platform.  You can either post it here if you don't mind or you can email it to me at jason@windscribe.com.

**Comment by blknight88** [comment]
No worries, just sent you an email, thanks 

I'll post the solution here, once we figure it out for others to see.

**Comment by jaxu** [comment]
Thanks, received.  I'll look at this probably Monday and reply here when I have a conclusion.

**Comment by blknight88** [comment]
Hey,

Sounds good, thanks!
________________________________

**Comment by jaxu** [comment]
For anyone lurking here, here's an update of the autoconnect issue from blknight88.

We've had a bit of an offline conversation where it came to light that autoconnect works fine, but only when the user it was installed for logs in.  It is killed when the user logs out.  This is sort of expected because the windscribe process runs as a user service, which is only valid during the user session.

One workaround for this is to enable linger for the user via `loginctl enable-linger <user>`.  This will make a user session spawn at boot, and will continue to exist if the user logs out.

We are also exploring other ways to make this work on our side and I've added a ticket to our board.

**Comment by jaxu** [comment]
I've found a few areas in the code which don't work quite right in pure CLI mode and is affecting the functionality of the autoconnect feature.  A fix should be available in an upcoming 2.13 build.

**Comment by blknight88** [comment]
Cool, what's a rough ETA for that build? Days, weeks, months?

Sincerely,

      Dominic Johnson

________________________________

**Comment by jaxu** [comment]
No guarantees but my guess is 1-2 weeks.

**Comment by blknight88** [comment]
Of course, oh ok not too bad thanks!

Sincerely,

      Dominic Johnson
________________________________

**Comment by bernerdad** [comment]
Upcoming build mentioned by @jaxu above is available now. [2.13.3 release](https://github.com/Windscribe/Desktop-App/releases/tag/v2.13.3)

**Comment by blknight88** [comment]
I can confirm that using the new alpha 2.13.3 worked for fixing autoconnect for me. Thanks for all the help!

The only other thing missing in this thread is due to my instance being a server, I had to use this command 
`loginctl enable-linger <user>` to get it to work on reboot without me logging in every time.

**Comment by jaxu** [comment]
> I can confirm that using the new alpha 2.13.3 worked for fixing autoconnect for me. Thanks for all the help!
> 
> The only other thing missing in this thread is due to my instance being a server, I had to use this command `loginctl enable-linger <user>` to get it to work on reboot without me logging in every time.

Thanks for confirming.  That command is about half a dozen message above.  I'll try to find some suitable place to add documentation for this as well.

**Comment by VityaSchel** [comment]
Any solution? AllowLANTraffic=true does not work for me

**Comment by blknight88** [comment]
Did you run `windscribe-cli preferences reload` after setting `AllowLANTraffic=true`?

This is how I did it and it works fine with version 2.13.8:

```
nano ~/.config/Windscribe/windscribe_cli.conf

  AllowLANTraffic=true

windscribe-cli preferences reload

# Turn on Firewall and connect 
windscribe-cli firewall on
windscribe-cli connect
windscribe-cli status
```

**Comment by VityaSchel** [comment]
> Did you run `windscribe-cli preferences reload` after setting `AllowLANTraffic=true`?
> 
> This is how I did it and it works fine with version 2.13.8:
> 
> ```
> nano ~/.config/Windscribe/windscribe_cli.conf
> 
>   AllowLANTraffic=true
> 
> windscribe-cli preferences reload
> 
> # Turn on Firewall and connect 
> windscribe-cli firewall on
> windscribe-cli connect
> windscribe-cli status
> ```

Yeah it doesn't work
I ended up adding some very simple basic rules to ip route command and switching to openvpn with windscribe profiles and it works perfect. In addition to that I can also specify ips to split tunnel in ovpn file

**Comment by blknight88** [comment]
Forgot I have a static route setup in `/etc/netplan/00-installer-config.yaml` on my server's NIC so my local traffic knows to use my local Gateway for my subnet. That could be the other missing piece as to why mine works. 

Either way glad you got it working!

**Comment by qomhmd** [comment]
I, obviously, was not able to connect to my **remote** server via SSH by setting `AllowLANTraffic` to `true` on it. Is there a way?

P.S. Turning off the firewall feature using `windscribe-cli firewall off` before connecting works.
